src/crypto/verification/Base.js

```diff
@@ -60,9 +63,25 @@ export default class VerificationBase extends EventEmitter {
         this.transactionId = transactionId;
         this.startEvent = startEvent;
         this.request = request;
+        this.cancelled = false;
         this._parent = parent;
         this._done = false;
         this._promise = null;
+        this._transactionTimeoutTimer = null;
+
+        // At this point, the verification request was received so start the timeout timer.
+        this._pingTransaction();
+    }
+
+    _pingTransaction() {
+        console.log("Refreshing/starting the verification transaction timeout timer");
+        clearTimeout(this._transactionTimeoutTimer);
```

Comment:
I assume the browser is fine with you passing null here, but probably worth a null check anyway.

Reply:
It's very fine with it. The JS spec even says it should silently do nothing for invalid handles.

I'll add a null check though